packages/react-router/lib/dom/ssr/components.tsx

```diff
@@ -279,7 +279,6 @@ export function PrefetchPageLinks({
   );
 
   if (!matches) {
-    console.warn(`Tried to prefetch ${page} but no routes matched.`);
```

Comment:
I think this is fine to remove entirely as if the route doesn't truly exist it'll 404 anyway so there's no real benefit in "hey I cant prefetch anything for this link that won't work anyway".  

If we really wanted to keep it, then I think we would need to add some verbiage around the fact that it might print once prior to the route being discovered but then work after discovery completes, which is gross.